handbook/engineering/web/goals.md

```diff
@@ -1,22 +1,94 @@
 # Web Team Goals and Priorities 
 
-## Goals
-
-### Long term
+**_The web team has three areas of focus:_**
 
-**_The web team has four areas of focus:_**
-
-  1. **_Creating and maintaining a highly usable and intentionally designed webapp interface_**
   1. **_Delivering the full, unique value of Sourcegraph [extensions](https://docs.sourcegraph.com/extensions)_**
   1. **_Maintaining and expanding_ [_code host integrations_](https://docs.sourcegraph.com/dev/background-information/web/code_host_integrations)**
+  1. **_Creating and maintaining a highly usable and intentionally designed webapp interface_**
```

Comment:
@Joelkw Is there an opportunity here to start speaking to the two workstreams (which will eventually become their own teams) outside of Code Insights namely: Frontend Platform and Integrations?

Since we are planning on having a level of operational divide between these workstreams it might make sense to let this divide take shape in our goals as well?

Copying the responsibilities that we defined as part of the eng org workshop for these two workstreams here: https://docs.google.com/document/d/1YcGzp2deYAMZnuGGe2hnA1Lc02polkNv0QVyu8Rm6jg/edit#bookmark=id.bf55h6fj2cyi


```
Integrations
  - Sourcegraph extensions: bringing the powers of other dev tools to Sourcegraph
    - Sourcegraph extension API
    - Extensions we write to bring the value of other dev tools to Sourcegraph
    - Docs/tutorials that enable 100+ third party extensions that are actively used.
  - Integrations: Bringing the power of Sourcegraph to other dev tools
    - Sourcegraph editor extensions 
      - Code intelligence in the editor powered by Sourcegraph
      - Search results in the editor
      - Sourcegraph extensions flow through to code I’m writing right now
    - Website integrations 
      - Browser extension supports multiple Sourcegraph instances
      - Browser extension architecture
        - Concerns around security, loading of Sourcegraph extensions (we had to workaround this for Firefox).
      - Direct integrations with code hosts (e.g. GitLab)
      - Direct integrations with sites where developers read code (e.g., https://pkg.go.dev, https://reactjs.org/, https://stackoverflow.com)

Frontend platform and code browsing
- Platform: How we do web development at Sourcegraph
  - Sourcegraph UI component library
  - Sourcegraph web tech stack, tools, patterns, documentation, education (e.g. webpack, TypeScript, React, RxJS)
  - Define and maintain how we test frontend code
  - Application UI (re)design
- Code browsing: Core user experience of Sourcegraph product
  - Site navigation and information hierarchy
  - Syntax highlighting + blob view
  - Repository landing pages and file directory pages
  - Basic code host like features (e.g. view diffs, commits, history)
  - Sourcegraph application homepage
  - Accessibility support

```

Reply:
@jeanduplessis I think that's a good idea. I've defined as such, with a couple notes: 
- I didn't give each goals type [frontend/integrations] its own subsection purely because if I made the structure Goals 
  \\-- Extensibility
      \\\\--- Make Sourcegraph extensions do...
      \\\\--- Make IDE extension... 
  \\-- Frontend
then goals end up being h4's and they weren't visually distinct enough to quickly scan. Experimented with other solutions but also happy to return to that defined split (rather than the "(tagging)" in the title I went with). 

- I want to strongly suggest we don't use the word "integrations" for this workstream and instead use something like "extensibility." For my reasoning, please see point 2 in [this comment](https://github.com/sourcegraph/about/pull/2153#issuecomment-742199652)